src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1352,31 +1609,46 @@ internal ProcessInputWriter(InternalCommand command)
         {
             Dbg.Assert(command != null, "Caller should validate the parameter");
             _command = command;
+
+            _pipeline = ScriptBlock.Create("Out-String -Stream").GetSteppablePipeline();
+            _pipeline.Begin(true);
         }
```

Comment:
With your change to `ProcessInputWriter`, we won't be able to use the same InputWriter type to handle byte stream transfers between native commands -- we will need a thread to read byte output from upstream native command and write to the downstream command.
My opinion is to keep the "use-separate-thread" module of `ProcessInputWriter`, but again use a `BlockingCollection` between `Add` and the thread that does the actual write operation. It's essentially a consumer/provider problem.